File: JULY-2025/09-07-2025/p3439.go
func maxFreeTime(eventTime int, k int, startTime []int, endTime []int) int {
    n := len(startTime)
    gap := startTime[0]
    for i:=1; i<k; i++ { gap += startTime[i] - endTime[i-1] }
    if k < n { gap += startTime[k] - endTime[k-1] } else {
        gap += eventTime - endTime[n-1]
        return gap
    }
    max := gap
    for i:=k+1; i<n; i++ {
        if i == k+1 { gap -= startTime[0] } else {
            gap -= startTime[i-k-1] - endTime[i-k-2]
        }
        gap += startTime[i] - endTime[i-1]
        if max < gap { max = gap }
    }
    if n-k == 1 { gap -= startTime[0] } else {
        gap -= startTime[n-k-1] - endTime[n-k-2]
    }
    gap += eventTime - endTime[n-1]
    if max < gap { return gap }
    return max
}

File: JULY-2025/09-07-2025/p3439.cs
public class Solution {
    public int MaxFreeTime(int eventTime, int k, int[] startTime, int[] endTime) {
        var freeTime = new List<int>();
        int currEnd = 0,n=startTime.Length;
        for(int i=0;i<n;i++){
            freeTime.Add(startTime[i]-currEnd);
            currEnd=endTime[i];
        }
        freeTime.Add(eventTime-endTime[n-1]);

        //Sliding wind
        int l=0,res=0,curr=0;
        for(int r=0;r<freeTime.Count;r++){
            if(r-l+1>k+1){
                curr -= freeTime[l];
                l++;
            }
            curr += freeTime[r];
            res = Math.Max(res,curr);
        }
        return res;
    }
}